tests/integration/lib/helpers.py

```diff
@@ -166,3 +167,19 @@ async def run_pod_with_args(kube_client: AsyncClient, namespace, image_name, pod
         return log_lines
     finally:
         await kube_client.delete(Pod, name=pod.metadata.name, namespace=namespace)
+
+
+async def wait_for_all_replicaset_replicas_ready(kube_client: AsyncClient, namespace: str):
+    timeout = 30
+    start_time = time.time()
+    now = time.time()
+    while start_time + timeout > now:
+        now = time.time()
+        await asyncio.sleep(0.5)
+        async for rs in kube_client.list(ReplicaSet, namespace=namespace):
+            if not rs.status or not rs.status.readyReplicas:
+                break
+            if rs.status.readyReplicas != rs.status.replicas:
+                break
```

Comment:
Isn't this the wrong way around. It is going to exit the loop if no `readyReplicas` or the `readyReplicas != replicas` ?

Reply:
Ah right. I was focusing on the `while` loop and not really seeing the `async for` loop.

EMORECOFFEE